backends/mlu/kernels/funcs/mlu_baseop.cc

```diff
@@ -1506,22 +1534,72 @@ MLURNNDesc::~MLURNNDesc() {
   float alpha = 1.0f;
   float beta = 0.0f;
 
-  PADDLE_ENFORCE_MLU_SUCCESS(cnnlMatMul(handle,
-                                        transpose_a,
-                                        transpose_b,
-                                        reinterpret_cast<void*>(&alpha),
-                                        in0_desc,
-                                        in0,
-                                        in1_desc,
-                                        in1,
-                                        reinterpret_cast<void*>(&beta),
-                                        output_desc,
-                                        output));
+  // Create and set matmul descriptor
+  cnnlMatMulDescriptor_t matmul_desc;
+  PADDLE_ENFORCE_MLU_SUCCESS(cnnlMatMulDescCreate(&matmul_desc));
+  const int allow_tf32 = AllowMLUTF32() ? 1 : 0;
+  PADDLE_ENFORCE_MLU_SUCCESS(cnnlSetMatMulDescAttr(
+      matmul_desc, CNNL_MATMUL_ALLOW_TF32, &(allow_tf32), sizeof(int)));
+
+  int transpose_a_int = static_cast<int>(transpose_a);
+  PADDLE_ENFORCE_MLU_SUCCESS(cnnlSetMatMulDescAttr(
+      matmul_desc, CNNL_MATMUL_DESC_TRANSA, &(transpose_a_int), sizeof(int)));
+  int transpose_b_int = static_cast<int>(transpose_b);
+  PADDLE_ENFORCE_MLU_SUCCESS(cnnlSetMatMulDescAttr(
+      matmul_desc, CNNL_MATMUL_DESC_TRANSB, &(transpose_b_int), sizeof(int)));
+
+  // Create and get matmul algorithim
+  cnnlMatMulAlgo_t algo = nullptr;
+  PADDLE_ENFORCE_MLU_SUCCESS(cnnlMatMulAlgoCreate(&algo));
+
+  // Get workspace
+  size_t workspace_size;
+
+  cnnlMatMulHeuristicResult_t result = nullptr;
+  PADDLE_ENFORCE_MLU_SUCCESS(cnnlCreateMatMulHeuristicResult(&result));
+  int requested_algo_count = 1, return_algo_count = 0;
+  PADDLE_ENFORCE_MLU_SUCCESS(cnnlGetMatMulAlgoHeuristic(handle,
+                                                        matmul_desc,
+                                                        in0_desc,
+                                                        in1_desc,
+                                                        output_desc,
+                                                        output_desc,
+                                                        nullptr,
+                                                        requested_algo_count,
+                                                        &result,
+                                                        &return_algo_count));
+  PADDLE_ENFORCE_MLU_SUCCESS(
+      cnnlGetMatMulHeuristicResult(result, algo, &workspace_size));
+  auto& dev_ctx = GetDevCtxFromCTX(ctx);
+  Tensor workspace = ctx.AllocateTmpTensor<int8_t, MLUDeviceContext>(
+      {static_cast<int64_t>(workspace_size)}, dev_ctx);
+  void* workspace_ptr = workspace.mutable_data(ctx.GetPlace());
```

Comment:
use `Alloc` instead of `mutable_data`, check workspace allocation method in other Op for [example](https://github.com/PaddlePaddle/PaddleCustomDevice/blob/develop/backends/mlu/kernels/funcs/mlu_baseop.cc#L944).